Fase2

alvaroprado199103 opened the pull request.

alvaroprado199103 committed:
Fase 2

alvaroprado199103 committed:
Fase 2

alvaroprado199103 committed:
Fase 2 Fixed

alvaroprado199103 committed:
Fase 2 Fixed 2

alvaroprado199103 committed:
Fase 2 third fix

alvaroprado199103 committed:
TEst 4

alvaroprado199103 committed:
Holmes Changes

alvaroprado199103 committed:
Holmes changes

alvaroprado199103 committed:
Holmes Changes 3

alvaroprado199103 committed:
Holmes changes 4

alvaroprado199103 committed:
Changes Holmes 5

alvaroprado199103 committed:
Holmes Changes 6

alvaroprado199103 committed:
Holmes Changes 7

alvaroprado199103 committed:
Holmes Changes Final

alvaroprado199103 committed:
Fixing TC execution

alvaroprado199103 committed:
Fixing TC 2

alvaroprado199103 committed:
Fixing TC 3

alvaroprado199103 committed:
Fixing TC 4

alvaroprado199103 committed:
Fixing TC5

alvaroprado199103 committed:
Fixing TC 6

alvaroprado199103 committed:
Fixing TC 7

alvaroprado199103 committed:
Fixing TC 8

alvaroprado199103 committed:
Fixing TC 9

alvaroprado199103 committed:
Fix TC 1

alvaroprado199103 committed:
Fix Chrome

alvaroprado199103 committed:
Fix NV 1

alvaroprado199103 committed:
fix nv 3

alvaroprado199103 committed:
Fix NV 4

alvaroprado199103 committed:
fix nv 5

alvaroprado199103 committed:
Fix NV 6

alvaroprado199103 committed:
fix nv 7

alvaroprado199103 committed:
fix nv 8

alvaroprado199103 committed:
fix nv9

alvaroprado199103 committed:
fix nv 10

alvaroprado199103 committed:
Fix nv11

alvaroprado199103 committed:
Fix nv 12

alvaroprado199103 committed:
Fix nv 13

alvaroprado199103 committed:
Fix nv 14

alvaroprado199103 committed:
Fix nv 15

alvaroprado199103 committed:
fix NV 16 quitando docker

alvaroprado199103 committed:
Fix nv 17

alvaroprado199103 committed:
Fix nv 18

holgiosalos commented on .github/workflows/gradle.yml:
```diff
@@ -1,67 +1,81 @@
+  push:  
+    branches: [ "main" ]  
+  pull_request:  
+    types: [opened, synchronize, reopened]  
```
Seria mejor configurarlo de la siguiente manera, teniendo en cuenta que normalmente creamos las branches con el prefijo feature/:
```
  push:
    branches:
      - main
      - 'feature/**' 
  pull_request:
    branches:
      - main
```

holgiosalos commented on .github/workflows/gradle.yml:
```diff
@@ -1,67 +1,81 @@
+jobs:  
+  build:  
+    runs-on: ubuntu-latest  
+    permissions:  
+      contents: read  
```
no estoy seguro si esto es necesario, por que lo agregaste?

holgiosalos commented on .github/workflows/gradle.yml:
```diff
@@ -1,67 +1,81 @@
+          ~/.gradle/wrapper  
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}  
+        restore-keys: |  
+          ${{ runner.os }}-gradle-  
```
Creo que este paso no es necesario, tienes una razon del porque es importante?

holgiosalos commented on .github/workflows/gradle.yml:
```diff
@@ -1,67 +1,81 @@
 
-  dependency-submission:
+    - name: Grant execute permission for gradlew  
+      run: chmod +x gradlew  
```
probablemente no necesario si borras la seccion de permissions

holgiosalos commented on .github/workflows/gradle.yml:
```diff
@@ -1,67 +1,81 @@
-    # See: https://github.com/gradle/actions/blob/main/dependency-submission/README.md
-    - name: Generate and submit dependency graph
-      uses: gradle/actions/dependency-submission@af1da67850ed9a4cedd57bfd976089dd991e2582 # v4.0.0
+    - name: Run Checkstyle  
+      run: ./gradlew checkstyleMain checkstyleTest  
+
+    - name: Upload Checkstyle report  
+      uses: actions/upload-artifact@v4  
+      with:  
+        name: checkstyle-report  
+        path: build/reports/checkstyle  
```
Como las pruebas estaticas se ejecutan mas rapido, y verifican buenas practicas, deberian ejecutarse antes de ejecutar las pruebas

holgiosalos commented on .github/workflows/gradle.yml:
```diff
@@ -1,67 +1,81 @@
+        java-version: '22'  
+        distribution: 'temurin'  
+    - name: Generate and submit dependency graph  
+      uses: gradle/actions/dependency-submission@af1da67850ed9a4cedd57bfd976089dd991e2582  
```
no estoy seguro si esto es necesario, dado que no estamos creando una dependencia o un artefacto. Por qué consideras que es necesario?

holgiosalos commented on bin/test/.DS_Store:
nunca debemos subir archivos .DS_Store, ni tampoco es comun que se deba subir la carpeta /bin

holgiosalos commented on bin/test/pages/BasePage.class:
No debemos subir archivos .class.

holgiosalos commented on bin/test/TCS.feature:
este archivo no deberia estar, la carpeta bin deberia estar en el .gitignore

holgiosalos commented on build.gradle:
```diff
@@ -1,55 +1,64 @@
+    implementation 'io.github.bonigarcia:webdrivermanager:5.9.3'  
+    testImplementation 'io.rest-assured:rest-assured:5.4.0'  
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.13.1'  
+    // Añadir la dependencia slf4j-simple  
```
evitar los comentarios en español

holgiosalos commented on config/checkstyle/checkstyle.xml:
```diff
@@ -0,0 +1,52 @@
+    <!-- Ensure a newline at the end of the file -->
+    <module name="NewlineAtEndOfFile"/>
+
+</module>
```
Evita subir archivos sin el EOL. Es mas o menos un standard en la industria que todos los archivos finalicen con una linea vacia al final del archivo, por esa razon GitHub alerta cuando algun archivo no lo hace con este simbolo 
![image](https://github.com/user-attachments/assets/dda08b47-70a9-418d-a9e9-9de79d4c8c9b)

holgiosalos commented on src/test/java/pages/BasePage.java:
```diff
@@ -6,10 +6,13 @@
 public class BasePage {
 
     protected WebDriver driver;
+    {
+
+    }
```
ibas a declarar algo aqui?

holgiosalos commented on src/test/java/runner/Scenario1.java:
Evita subir comentarios en español

holgiosalos commented on src/test/java/runner/Scenario1.java:
```diff
@@ -12,4 +11,4 @@
 
 public class Scenario1 {
```
Podrias usar un mejor nombre, algo asi como CucumberRunner o ScenariosRunner

holgiosalos commented on src/test/java/runner/Scenario2.java:
Creeria que este runner no es necesario

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -20,15 +21,33 @@ public class SauceDemoSteps {
+        System.out.println("Inicializando el WebDriver...");
+        try {
+            ChromeOptions options = new ChromeOptions();
+            options.addArguments("--headless"); // Ejecutar sin interfaz gráfica
+            options.addArguments("--disable-gpu"); // Mejor compatibilidad en algunos entornos
+            options.addArguments("--no-sandbox"); // Evita problemas en entornos CI/CD
+            options.addArguments("--disable-dev-shm-usage"); // Mejora estabilidad en contenedores
```
evita los comentarios en español

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -20,15 +21,33 @@ public class SauceDemoSteps {
-        driver = new ChromeDriver();
-        loginPage = new LoginPage(driver);
-        mainPage = new MainPage(driver);
+        System.out.println("Inicializando el WebDriver...");
```
Evita el uso de System.out.println, en su lugar debes usar un Logger

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -20,15 +21,33 @@ public class SauceDemoSteps {
-        mainPage = new MainPage(driver);
+        System.out.println("Inicializando el WebDriver...");
+        try {
+            ChromeOptions options = new ChromeOptions();
+            options.addArguments("--headless"); // Ejecutar sin interfaz gráfica
+            options.addArguments("--disable-gpu"); // Mejor compatibilidad en algunos entornos
+            options.addArguments("--no-sandbox"); // Evita problemas en entornos CI/CD
+            options.addArguments("--disable-dev-shm-usage"); // Mejora estabilidad en contenedores
+            System.out.println("Configurando WebDriverManager...");
+            WebDriverManager.chromedriver().setup();
+            System.out.println("WebDriverManager configurado.");
+            System.out.println("Creando instancia de ChromeDriver...");
+            driver = new ChromeDriver(options);
```
Mueve toda la logica de configurar e inicializar el webdriver a una nueva clase. Algo asi como WebDriverFactory, haz que implemente un factory method que se llame createChromeDriver

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -20,15 +21,33 @@ public class SauceDemoSteps {
     @After
     public void closeNav() {
-        driver.close();
+        // driver.quit();
```
no subas codigo comentado

holgiosalos reviewed (changes requested):
Por favor aplica estos cambios generales:

- Elimina la carpeta bin y cualquier archivo .class y .DS_Store
- Luego que elimines los archivos en un nuevo commit configura el gitignore de manera que se sigan ignorando a futuro estos archivos.

holgiosalos commented on config/checkstyle/checkstyle.xml:
Intentemos usar la gia de estilo de google, de esta manera podemos aderirnos a uno de los estandares de la industria:
https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/google_checks.xml

holgiosalos replied:
Recuerda este comentario, aqui te sugiero usar una base de reglas mas amplias que nos ayude con otro conjunto de buenas practicas

holgiosalos commented on src/test/resources/TCS.feature:
```diff
@@ -3,11 +3,13 @@ Feature: Navigation bar
     Without logging in
     I can the login workflow
```
Para la descripcion del feature file deberiamos ser un poco mas claros, algo asi como:
```
Feature: Login for SauceDemo page
    Validate behavior when login using valid and invalid credentials for standard and locked users
```

holgiosalos commented on src/test/resources/TCS.feature:
```diff
@@ -3,11 +3,13 @@ Feature: Navigation bar
     Without logging in
     I can the login workflow
 
+    @test
```
No considero necesaria la anotacion por el momento

holgiosalos commented on src/test/resources/TCS.feature:
```diff
@@ -3,11 +3,13 @@ Feature: Navigation bar
     I can the login workflow
 
+    @test
     Scenario: I can try to login using a correct and wrong password.
         Given I navigate to www.saucedemo.com
         When Login using a correct password
         Then Login using incorrect password
```
Seria mejor dividir los scenarios, uno para cuando se usan credenciales correctas y otro para cuando se usan credenciales incorrectas

holgiosalos replied:
Y en el Then finalmente validamos el comportamiento descrito en el when, "Then the user should be able to access to the inventory page"

holgiosalos commented on src/test/resources/TCS.feature:
```diff
@@ -3,11 +3,13 @@ Feature: Navigation bar
 
+    @test
     Scenario: I can try to login using a correct and wrong password.
         Given I navigate to www.saucedemo.com
```
Puedes usar Background dado que este given es comun a todos los scenarios

holgiosalos commented on src/test/resources/TCS.feature:
```diff
@@ -3,11 +3,13 @@ Feature: Navigation bar
+    @test
     Scenario: I can try to login using a correct and wrong password.
         Given I navigate to www.saucedemo.com
         When Login using a correct password
```
El when puede ser mas descriptivo y asegurarse que solo se encargue de loguear al usuario. Por ejemplo  "When standard user logs in with valid credentials"

holgiosalos commented on src/test/resources/TCS.feature:
De manera general, te recomiendo:

Uso del background para el login.

Tener 3 scenarios, 1 para login con credenciales correctas, otro para login con credenciales incorrectas y el ultimo para el login con el usuario bloqueado. 

En cada scenario la idea es que el when solo realice la accion de login y el then verifique el resultado esperado. Ten en cuenta que al cambiar la descripcion y el objetivo de cada step, debes actualizar la case de step definitions.

Recomiendo revisar las siguientes paginas para tener mas claridad cuando se diseñan casos de pruebas usando Gherkin:
- https://cucumber.io/docs/bdd/better-gherkin/
- https://www.browserstack.com/guide/cucumber-best-practices-for-testing

holgiosalos commented:
Finalmente, actualiza las configuraciones de la rama para que no permita hacer merge de los pull request cuando se requieren cambios:
Ir a Settings > Branches adicionamos una regla dando click en add rule. Escribimos main en el campo de branch name pattern. Una vez hecho eso, damos click en las siguientes opciones:
![image](https://github.com/user-attachments/assets/ba74bcbb-28d6-4e6c-b24d-d6a21b7ec775)

alvaroprado199103 committed:
feat: Update project structure and improve best practices

- Refactored GitHub Actions workflows for better CI/CD integration.
- Enhanced dependency management in build.gradle with updated versions.
- Adopted Google Style Guide for Checkstyle configuration.
- Improved feature files for clarity and separation of concerns in Cucumber scenarios.
- Implemented feedback regarding naming conventions and code organization.
- Established repository settings to require pull request reviews before merging.

alvaroprado199103 committed:
feat: Update project structure and improve best practices

- Refactored GitHub Actions workflows for better CI/CD integration.
- Enhanced dependency management in build.gradle with updated versions.
- Adopted Google Style Guide for Checkstyle configuration.
- Improved feature files for clarity and separation of concerns in Cucumber scenarios.
- Implemented feedback regarding naming conventions and code organization.
- Established repository settings to require pull request reviews before merging.

holgiosalos commented on src/test/java/runner/CucumberRunner.java:
```diff
@@ -0,0 +1,12 @@
+
+@RunWith(Cucumber.class)
+@CucumberOptions(features = "src/test/resources", glue = "steps", plugin = { "pretty", "html:target/cucumber-reports" })
+
```
remove this blank line

alvaroprado199103 replied:
Fixed

alvaroprado199103 replied:
done

holgiosalos commented on bin/test/steps/SauceDemoSteps.class:
este archivo no puede estar presente, elimina la carpeta bin

alvaroprado199103 replied:
se eliminó usando las siguientes lineas de codigo: git rm -r --cached bin
git rm --cached bin/test/.DS_Store
git commit -m "Eliminar bin y .DS_Store del repositorio"

alvaroprado199103 replied:
se borro usando comandos git

holgiosalos commented on bin/test/pages/BasePage.class:
este archivo no puede estar presente, elimina la carpeta bin

alvaroprado199103 replied:
se eliminó usando las siguientes lineas de codigo: git rm -r --cached bin
git rm --cached bin/test/.DS_Store
git commit -m "Eliminar bin y .DS_Store del repositorio"

holgiosalos commented on bin/test/.DS_Store:
este archivo no puede estar presente, elimina la carpeta bin

holgiosalos commented on .github/workflows/gradle.yml:
```diff
@@ -1,67 +1,56 @@
+        path: build/reports/checkstyle  
+    
+    - name: Run Tests  
+      run: ./gradlew test --tests "runner.CucumberRunner"  
```
archivo no finaliza con una linea en blanco

alvaroprado199103 replied:
Fixed

holgiosalos commented on src/test/java/utilities/WebDriverFactory.java:
```diff
@@ -0,0 +1,68 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class WebDriverFactory {
```
La responsabilidad de un esta clase radica solo en crear objetos, no deberias tener "estado" es decir no deberian existir variables privadas, tampoco deberian existir metodos que no sean exclusivamente crear objetos de tipo WebDriver.

alvaroprado199103 replied:
Fixed.

holgiosalos commented on src/test/java/utilities/WebDriverFactory.java:
```diff
@@ -0,0 +1,68 @@
+
+    private static final Logger LOGGER = Logger.getLogger(WebDriverFactory.class.getName());
+    private static final int TIMEOUT_SECONDS = 10;
+    private WebDriver driver;
```
no deberia existir este objeto dentro de la  clase

alvaroprado199103 replied:
ya fue puesto dentro de la clase que lo retorna

holgiosalos commented on src/test/java/utilities/WebDriverFactory.java:
```diff
@@ -0,0 +1,68 @@
+    private WebDriver driver;
+
+    public WebDriver createDriver() {
+        LOGGER.log(Level.INFO, "Initializing WebDriver...");
```
dentro de este metodo es donde debes crear el webdriver y retornarlo

alvaroprado199103 replied:
fixed

holgiosalos commented on src/test/java/utilities/WebDriverFactory.java:
```diff
@@ -0,0 +1,68 @@
+        return driver;
+    }
+
+    public void closeDriver() {
+        if (driver != null) {
+            LOGGER.log(Level.INFO, "Closing WebDriver...");
+            driver.quit();
+            LOGGER.log(Level.INFO, "WebDriver closed.");
+        }
+    }
```
este metodo no es responsabilidad de esta clase, por ende no deberia existir

alvaroprado199103 replied:
se agrego en la clase utils

holgiosalos commented on src/test/java/utilities/WebDriverFactory.java:
```diff
@@ -0,0 +1,68 @@
+        }
+    }
+
+    public WebDriver getDriver() {
+        return driver;
+    }
```
este metodo no es responsabilidad de esta clase, por ende no deberia existir

alvaroprado199103 replied:
eliminado

holgiosalos commented on src/test/java/utilities/WebDriverFactory.java:
```diff
@@ -0,0 +1,68 @@
+        return driver;
+    }
+
+    // Explicit wait for a WebElement to be visible
+    public void waitForVisibility(WebElement element) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(TIMEOUT_SECONDS));
+        wait.until(ExpectedConditions.visibilityOf(element));
+    }
+
+    // Explicit wait for a WebElement to be clickable
+    public void waitForClickability(WebElement element) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(TIMEOUT_SECONDS));
+        wait.until(ExpectedConditions.elementToBeClickable(element));
+    }
```
estos metodos no son responsabilidad de esta clase, por ende deberias crear una nueva clase con estos dos metodos, puede ser una clase de Utils para las esperas, algo como WebDriverWaitUtils, y en el constructor inyectar (recibir) el driver

alvaroprado199103 replied:
se creo dicha clase

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -1,60 +1,68 @@
-        mainPage = new MainPage(driver);
+        webDriverFactory = new WebDriverFactory();
+        driver = webDriverFactory.createDriver();
+        loginPage = new LoginPage(driver, webDriverFactory);
+        mainPage = new MainPage(driver, webDriverFactory);
```
en lugar de inyectar el webdriverFactory en las clases login y main page, podrias hacer que dentro de estas clases se inicialice la nueva clase util de WebDriverWaitUtils

alvaroprado199103 replied:
done

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -1,60 +1,68 @@
+    @When("a blocked user attempts to log in")
+    public void loginBlockedAccount() {
         loginPage.loginUser("locked_out_user", "secret_sauce");
         Assert.assertTrue(loginPage.getErrorMessageB());
```
este assertion no deberia ejecutarse aqui, ya que ya lo haces en el then

alvaroprado199103 replied:
eliminado.

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -1,60 +1,68 @@
     }
 
+    @Then("the user should be able to access the inventory page")
+    public void userShouldAccessInventory() {
```
este metodo deberia tener algun assertion, puedes crear un page object para la pagina de inventory pero solo con un elemento, alguno que te indique que la pagina abrio correctamente

alvaroprado199103 replied:
se agrego un asertion relacionado al titulo del main page.

alvaroprado199103 committed:
Eliminar bin y .DS_Store del repositorio

alvaroprado199103 committed:
Feat: Update project structure and improve best practices

- Refactored GitHub Actions workflows for better CI/CD integration.
- Enhanced dependency management in build.gradle with updated versions.
- Adopted Google Style Guide for Checkstyle configuration.
- Improved feature files for clarity and separation of concerns in Cucumber scenarios.
- Implemented feedback regarding naming conventions and code organization.
- Established repository settings to require pull request reviews before merging.
- Refactored SauceDemoSteps class to use WebDriverWaitUtils for better synchronization.
- Removed redundant assertions to follow BDD best practices in Cucumber steps.
- Updated LoginPage and MainPage to use WebDriverWaitUtils instead of WebDriverFactory.
- Fixed incorrect credentials in invalid login step for more accurate testing.
- Cleaned up Git tracking to ignore bin folder and unnecessary files.
- Improved XPath selectors to ensure robustness against UI changes.

holgiosalos commented on bin/test/TCS.feature:
este archivo tambien lo debes eliminar porque esta dentro de la carpeta bin

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -1,60 +1,70 @@
     @After
     public void closeNav() {
-        driver.close();
+        webDriverWaitUtils.closeDriver();
```
como inicializas en el setup el driver que luego es reusado por todas las paginas, creo que lo correcto es cerrarlo y no encapsular ese cierre en el WebDriverWaitUtils

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -1,60 +1,70 @@
-    public void LoginFailed() {
-        loginPage.navigateToSauceDemo();
+    @When("standard user logs in with invalid credentials")
+    public void loginFailed() {
```
cambia el nombre del metodo por algo mas preciso como: userLogsInWithInvalidCredentials()

holgiosalos commented on src/test/java/utilities/WebDriverWaitUtils.java:
```diff
@@ -0,0 +1,40 @@
+            driver.quit();
+            LOGGER.log(Level.INFO, "WebDriver closed.");
+        }
+    }
```
este metodo no deberia ser responsabilidad de esta clase, creo que ademas no es necesario

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -1,60 +1,70 @@
-    public void LoginSuccesfully() {
-        loginPage.navigateToSauceDemo();
+    @When("standard user logs in with valid credentials")
+    public void loginSuccessfully() {
```
cambia el nombre del metodo por algo mas preciso como: userLogsInWithValidCredentials

holgiosalos commented on src/test/java/steps/SauceDemoSteps.java:
```diff
@@ -1,60 +1,70 @@
 
+    @Then("the user should be able to access the inventory page")
+    public void userShouldAccessInventory() {
+        mainPage.MainMenuValidation();
```
aqui deberiamos tener un assertion, y el metodo quizas llamarse distinto:
1. renombre el metodo como isMainMenuVisible y hazlo retornar true si el menu es visible. 
2. realiza aqui el assertion similar a como se hace en el metodo del mensaje de error

holgiosalos commented on src/test/java/pages/MainPage.java:
```diff
@@ -20,25 +16,32 @@ public class MainPage extends BasePage {
 
-    // Error MsgValidation
+    // Error Msg Validation
     public boolean ErrorMsg() {
-        wait.until(d -> ErrorMsg.isDisplayed());
+        webDriverWaitUtils.waitForVisibility(ErrorMsg);
         return ErrorMsg.isDisplayed();
     }
```
Creo que el metodo ErrorMsg en el main page no es necesario, y creo que no se esta usando en ningun lado.

holgiosalos commented on src/test/java/pages/MainPage.java:
```diff
@@ -20,25 +16,32 @@ public class MainPage extends BasePage {
     }
 
+    // Main menu Dashboard validation
+    public void MainMenuValidation() {
+        webDriverWaitUtils.waitForVisibility(MainMenuTitle);
+    }
```
Como lo menciono en un comentario en la clase de step defs, este metodo deberia retronar true o false dependiendo si el menu es visible, ademas, de llamarse distinto, por ejemplo isMainMenuVisible

holgiosalos commented on src/test/java/pages/LoginPage.java:
```diff
@@ -42,5 +41,4 @@ public void loginUser(String username, String password) {
     public boolean getErrorMessageB() {
```
renombre este metodo para que se llame isErrorMessageDisplayed, por lo general los metodos que retornan un boolean deben tener un nombre que sugiere que retorna true o false.

alvaroprado199103 committed:
feature: Deleting junk files and updating gitignore

alvaroprado199103 committed:
Feature: Fixing comments from line manager.

alvaroprado199103 committed:
Delete src/test/resources/fase3.feature

deletion File Fase 3

alvaroprado199103 commented:
### Phase 2: Pipeline Configuration (Current Phase)  

1. **Cross-Browser Testing:** Configure existing tests to run on multiple browsers.  
2. **GitHub Actions Integration:** Configure existing tests to run on GitHub Actions.  
   - **Triggers:**  
     - Scheduled execution once per day at a fixed time.  
     - Triggered upon creation of a new PR.  
     - Triggered upon merging into the main branch.  
   - **Static Analysis:** Enable linters or static code analysis tools.
